### PR TITLE
EntitlementManagement - Add Types, move catalogs and packages to 1.0 Stable

### PR DESCRIPTION
**Pull request by kaovd**
Bugfixes and make stable where possible parts of the EntitlementManagement scopes

ODataTypes are required to be passed on the UserSet Struct


```
> $ env TF_DEBUG=TRACE TF_LOG=TRACE go test -race ./msgraph -run="TestAccessPackage*"                                                                                                                                                                                                                                                                                   [±fix/identitygovernance ●]
ok      github.com/manicminer/hamilton/msgraph  143.400s
```

### CODE REVIEW

**Comment by kaovd** [comment]
I also need to add "#microsoft.graph.question"to questions Struct, will do this later

File: msgraph/models.go [hunk 1]
```diff
@@ -44,16 +41,14 @@ type AccessPackageAssignmentPolicy struct {
...
-	CatalogType         AccessPackageCatalogType   `json:"catalogType,omitempty"`
-	CreatedBy           *string                    `json:"createdBy,omitempty"`
-	CreatedDateTime     *time.Time                 `json:"createdDateTime,omitempty"`
-	Description         *string                    `json:"description,omitempty"`
-	DisplayName         *string                    `json:"displayName,omitempty"`
-	IsExternallyVisible *bool                      `json:"isExternallyVisible,omitempty"`
-	ModifiedBy          *string                    `json:"modifiedBy,omitempty"`
-	ModifiedDateTime    *time.Time                 `json:"modifiedDateTime,omitempty"`
+	ID                  *string                   `json:"id,omitempty"`
+	State               AccessPackageCatalogState `json:"state,omitempty"`
```
Comment (manicminer) [inline msgraph/models.go:45]: As above, could we keep both fields for now in order to maintain compatibility with the beta API? I'm assuming the beta docs are correct and the field is still called `catalogStatus` in the beta model?

File: msgraph/valuetypes.go [hunk 2]
```diff
@@ -22,10 +22,11 @@ func (s StringNullWhenEmpty) MarshalJSON() ([]byte, error) {
 	return json.Marshal(string(s))
 }
 
-type AccessPackageCatalogStatus = string
+type AccessPackageCatalogState = string
```
Comment (manicminer) [inline msgraph/valuetypes.go:25]: Per compatibility note, can we keep both these types until such time as the beta API is updated

File: odata/types.go [hunk 3]
```diff
@@ -40,6 +44,9 @@ type Type = string
 const (
 	TypeAccessPackage                               Type = "#microsoft.graph.accessPackage"
 	TypeAccessPackageAssignmentPolicy               Type = "#microsoft.graph.accessPackageAssignmentPolicy"
+	TypeAccessPackageQuestion                       Type = "#microsoft.graph.accessPackageQuestion"
+	TypeAccessPackageTextInputQuestion                       Type = "#microsoft.graph.accessPackageTextInputQuestion"
+	TypeAccessPackageMultipleChoiceQuestion                       Type = "#microsoft.graph.accessPackageMultipleChoiceQuestion"
```
Comment (manicminer) [inline odata/types.go:49]: Not sure why the linter didn't pick this up, but could we tidy up the spacing for these?

**Review by manicminer — CHANGES_REQUESTED** [review]
Hi @kaovd, thanks for working on this! I've made a couple of notes on compatibility between the 1.0 and beta APIs if you can consider these? There may be other divergences I have missed. If it's impractical to maintain compatibility (e.g. perhaps the beta API isn't working right) then that's fine, but for now if we can that'd be good.

**Comment by kaovd** [comment]
Seems the test for AppRoleAssignedToClient.Assign is having issues @manicminer - do you want to try fix within the PR or just do it in as seperate fix

**Comment by kaovd** [comment]
Have synced with main build now passing @manicminer

**Comment by kaovd** [comment]
AccessPackageResourceRequests tests are now failing for some reason likely may have changed MS side need to review

**Comment by kaovd** [comment]
```
> $ go test -race ./msgraph -run="AccessPackage*" -v                                                                                                              [±fix/identitygovernance ✓]
=== RUN   TestAccessPackageClient
--- PASS: TestAccessPackageClient (8.86s)
=== RUN   TestAccessPackageAssignmentPolicyClient
--- PASS: TestAccessPackageAssignmentPolicyClient (4.77s)
=== RUN   TestAccessPackageCatalogClient
--- PASS: TestAccessPackageCatalogClient (1.63s)
=== RUN   TestAccessPackageResourceRequestClient
--- PASS: TestAccessPackageResourceRequestClient (18.34s)
=== RUN   TestAccessPackageResourceRoleScopeClient
--- PASS: TestAccessPackageResourceRoleScopeClient (17.03s)
PASS
ok      github.com/manicminer/hamilton/msgraph  50.657s
```
Can you give these tests a run locally @manicminer - I am not quite sure what the issues is with the CI but tests seem broken, but this passes fine for me locally

**Review by manicminer — APPROVED** [review]
Hi @kaovd, thanks for working on fixing this up, and sorry for the delay in reviewing. The test failure looks like a race condition stemming from the group creation - adding a sleep after line 21 in accesspackageresourcerequest_test.go seems to mitigate but we can leave that for now in lieu of a more robust way to handle propagation delays.

This LGTM! 🚀

**Pull request merged**